### PR TITLE
Upgrade to version 3 of Terraform AWS provider

### LINKED ISSUE #7: Automate association of a VPC with a private DNS zone (cross-account)
Terraform is currently unable to automate the process of associating a VPC in one AWS account with a private DNS zone in a different AWS account.  This [feature request](https://github.com/terraform-providers/terraform-provider-aws/issues/384) details the issue.

This process can be done via the AWS CLI, so that is [our current manual workaround](https://github.com/cisagov/cool-assessment-terraform/blob/develop/vpc.tf#L40-L63):
```
aws route53 create-vpc-association-authorization \
--hosted-zone-id COOLPRIVATEZONEID \
--vpc VPCRegion=us-east-1,VPCId=vpc-MY_ENV_VPC_ID \
--profile cool-sharedservices-provisionaccount

aws route53 associate-vpc-with-hosted-zone \
--hosted-zone-id COOLPRIVATEZONEID \
--vpc VPCRegion=us-east-1,VPCId=vpc-MY_ENV_VPC_ID \
--profile cool-MYENV-provisionaccount
```

It may be possible to use a Terraform module (such as [this one](https://github.com/opetch/terraform-aws-cli-resource)) to accomplish this automatically until Terraform is able to natively perform this action.

### PR DESCRIPTION
**Pull request by dav3r**
# 

## 🗣 Description
This PR moves this repo from using version 2 to version 3 of the [Terraform AWS provider](https://github.com/terraform-providers/terraform-provider-aws).

I also took this opportunity to update the example code so that it can be successfully deployed in our COOL environment.  The main change was to create a new VPC and deploy into that, rather than the default VPC.  This was done because we now delete the default VPC in all of our COOL accounts.

## 💭 Motivation and Context
Version 3 of the AWS provider appears to have stabilized, so it seems like a good time for us to jump on board.

It will also allow us to resolve https://github.com/cisagov/cool-assessment-terraform/issues/7 by taking advantage of https://github.com/terraform-providers/terraform-provider-aws/pull/14215.

I will keep this PR as a draft until https://github.com/cisagov/cert-read-role-tf-module/pull/19 is merged.  After that, I will revert https://github.com/cisagov/openvpn-server-tf-module/commit/8041de9373349e46377356f4dc2c4bcda754c483, finalize this PR, and request review.

## 🧪 Testing
* All automated tests pass.
* The example code was successfully deployed and destroyed.

## 🚥 Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (causes existing functionality to change)

## ✅ Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: iam.tf [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # Create a role that allows the instance to read its certs from S3.
 
 module "certreadrole" {
-  source = "github.com/cisagov/cert-read-role-tf-module"
+  source = "github.com/cisagov/cert-read-role-tf-module?ref=test_tf_aws_provider_v3"
```
Comment (jsf9k) [inline iam.tf]: Don't forget to revert this change before merging.

**Review by jsf9k — APPROVED** [review]
Nice work!

File: iam.tf (hunk 1 above)
Comment (dav3r) [reply iam.tf]: From the **Motivation and Context** section above 😄 :
> I will keep this PR as a draft until cisagov/cert-read-role-tf-module#19 is merged. After that, I will revert 8041de9, finalize this PR, and request review.

Comment (dav3r) [reply iam.tf]: Reverted with https://github.com/cisagov/openvpn-server-tf-module/pull/48/commits/301c3edd590d9a9e1581da804f4ddc0de5f6e48e.

**Review by felddy — APPROVED** [review]
Looking good.  💪

**Review by hillaryj — APPROVED** [review]
🥞 Looks great! Thanks for updating the examples too.
